rewrite-core/src/main/java/org/openrewrite/TreeVisitor.java

```diff
@@ -102,7 +107,7 @@ public T visit(@Nullable Tree tree, P p) {
             cursor.set(new Cursor(cursor.get(), tree));
         }
 
-        @SuppressWarnings("unchecked") T t = visitEach((T) tree, p);
+        @SuppressWarnings("unchecked") T t = preVisit((T) tree, p);
```

Comment:
note we don't call `postVisit` directly in TreeVisitor `visit`, is this acceptable? Still _works_ because each TreeVisitor implementation should have every `visitX` call `postVisit` before return, but it's not in this main TreeVisitor lifecycle. Should it? I suppose it technically shouldn't matter since we aren't doing anything special with it in the same way we are with `preVisit`, but again, just calling it out

Reply:
Hmm. Is there a reason why we prefer to call `preVisit` and `postVisit` from every visit implementation rather than once here? Nothing jumps out at me to explain why we _wouldn't_ put it here...

Reply:
Why would you not put the call to postVisit in visit? I would think it would work similar to preVisit/visitEach, except be called after the accept call.

Reply:
i suppose there is a double `visitEach` right now, and i've kind of been wondering how things get hit twice.